viewing_party/party.py

```diff
@@ -0,0 +1,199 @@
+##Wave 1!!
+
+
+def create_movie(title,genre,rating):
+    
+    if bool(title)==True and bool(genre)==True and bool(rating)==True:
+        dict={"title":"Title A","genre":"Horror","rating":3.5}
+        return dict
+    else:
+        return None
+
+def add_to_watched(user_data,movie):
+   
+    #user_data={'watched':[{},{},{}]}
+    #movie={"title": "Title A", "genre": "Horror", "rating": 3.5}
+    watched=user_data['watched']
+    watched.append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    #user_data=dictionary with a key `"watchlist"`, 
+    #and a value which is a list of dictionaries representing the movies the user wants to watch
+    #user_data={'watchlist':[{}{}{}]}
+    watchlist=user_data['watchlist']
+    watchlist.append(movie)
+    return user_data
+  
+def watch_movie(user_data,title):
+       #user_data={'watchlist':[{}{}{}], 'watched':[{}{}{}]}
+    list=[]
+    to_watch_movies=user_data['watchlist']
+    for i in range(len(to_watch_movies)):
+      to_watch_movie=to_watch_movies[i]['title']
+      list.append(to_watch_movie)
+
+    if title in list:
+        #i have to remove and add the movie corresponding to this title
+        movie_index=list.index(title)
+        x=to_watch_movies.pop(movie_index)
+        already_watched_movie=user_data['watched']
+        already_watched_movie.append(x)
+        return user_data
+    else: 
+        return user_data
+
+## WAVE 2 !!!
+
+
+def get_watched_avg_rating(user_data):
+    #user_data={'watched'=[{},{},{}]}
+    list_of_ratings=[]
+    watchedlist=user_data['watched']
+    for i in range(len(watchedlist)):
+      rating=user_data['watched'][i]['rating']
+      list_of_ratings.append(rating)
+    if len(watchedlist)==0:
+      avg_rating=0.0
+    else:
+      avg_rating=sum(list_of_ratings)/len(watchedlist)
+    
+    return avg_rating
+
+
+def get_most_watched_genre(user_data):
```

Comment:
even though this a good implementation, using a frequency map could've saved you some lines of code. Here is a resource for frequency maps https://www.geeksforgeeks.org/counting-the-frequencies-in-a-list-using-dictionary-in-python/